roles/openshift_facts/library/openshift_facts.py

```diff
@@ -463,13 +463,9 @@ def set_sdn_facts_if_unset(facts, system_facts):
         # default MTU if interface MTU cannot be detected
         facts['node']['sdn_mtu'] = '1450'
 
-        for val in itervalues(system_facts):
-            if isinstance(val, dict) and 'mtu' in val:
-                mtu = val['mtu']
-
-                if 'ipv4' in val and val['ipv4'].get('address') == node_ip:
-                    facts['node']['sdn_mtu'] = str(mtu - 50)
-
+        if 'ansible_default_ipv4' in system_facts:
+            if system_facts['ansible_default_ipv4'].get('address') == node_ip and 'mtu' in system_facts['ansible_default_ipv4']:
```

Comment:
I don't see the difference with current code: all `system_facts` are iterated, eventually `val` would be `ansible_default_ipv4` and the same code would be executed, isn't it?

Reply:
A cleaner fix would be:
```diff
- if 'ipv4' in val and val['ipv4'].get('address') == node_ip:
+ if val.get('address') == node_ip: 
```

Reply:
Preivously, system_facts was computed like this in openshift_facts.py __init__  :
           # pylint: disable=too-many-function-args,invalid-name
            self.system_facts = ansible_facts(module, ['hardware', 'network', 'virtual', 'facter'])  # noqa: F405
            additional_facts = {}
            for (k, v) in self.system_facts.items():
                additional_facts["ansible_%s" % k.replace('-', '_')] = v
            self.system_facts.update(additional_facts)

Now system_facts comes from roles/openshift_facts/vars/main.yml from vars_openshift_facts_system_facts.
vars_openshift_facts_system_facts:
  ansible_default_ipv4: "{{ vars.ansible_default_ipv4 }}"
  ansible_nodename: "{{ vars.ansible_nodename }}"
  ansible_fqdn: "{{ vars.ansible_fqdn }}"
  ansible_product_name: "{{ vars.ansible_product_name }}"
  ansible_product_version: "{{ vars.ansible_product_version }}"
  ansible_virtualization_type: "{{ vars.ansible_virtualization_type }}"
  ansible_virtualization_role: "{{ vars.ansible_virtualization_role }}"
  ansible_system_vendor: "{{ vars.ansible_system_vendor }}"

Reply:
Would the following be cleaner ?:

```diff
-            if isinstance(val, dict) and 'mtu' in val:
-                mtu = val['mtu']
 
-                if 'ipv4' in val and val['ipv4'].get('address') == node_ip:
-                    facts['node']['sdn_mtu'] = str(mtu - 50)
+            if isinstance(val, dict):
+                if val.get('address') == node_ip:
+                    facts['node']['sdn_mtu'] = str(val.get('mtu') - 50)
```

Reply:
Yep, looks cleaner :+1: